dpx/src/dpx_pdfobj.rs

```diff
@@ -1156,31 +1129,40 @@ pub unsafe extern "C" fn pdf_set_string(
     };
 }
 /* Name does *not* include the /. */
-#[no_mangle]
-pub unsafe extern "C" fn pdf_new_name(mut name: *const i8) -> *mut pdf_obj {
+pub unsafe fn pdf_new_name(name: &str) -> *mut pdf_obj {
```

Comment:
`pdf_new_name` splitted on `pdf_new_name(name: &str)` and `df_copy_name(name: *const i8)`.
Use Rust allocator.